toolchain/check/eval.cpp

```diff
@@ -1689,6 +1689,76 @@ static auto MakeConstantForBuiltinCall(EvalContext& eval_context,
 
     case SemIR::BuiltinFunctionKind::PrintChar:
     case SemIR::BuiltinFunctionKind::PrintInt:
+    case SemIR::BuiltinFunctionKind::StringAt:{
```

Comment:
This has changed the behaviour of the other builtin functions that it's below in this switch, no?

Did you try running `file_test`?

Also, this change should come with some new tests (probably in check/ and in lower/ for using this new builtin.